modules/DGM/PSO.cpp

```diff
@@ -0,0 +1,215 @@
+//
+// Created by ahambasan on 22.02.20.
+//
+
+#include "PSO.h"
+#include "macroses.h"
+
+#include <iterator>
+#include <vector>
+#include <random>
+
+DirectGraphicalModels::PSO::PSO() = default;
+
+DirectGraphicalModels::PSO::PSO(const vec_float_t &vParams)
+        : m_nParams(vParams.size()),
+          m_vParams(vParams.size()),
+          m_vMin(vParams.size()),
+          m_vMax(vParams.size()),
+          isThreadsEnabled(false) {
+    for (size_t p = 0; p < vParams.size(); p++) {
+        const float &param = vParams[p];
+        m_vParams[p] = param;
+    }
+    m_nParams = vParams.size();
```

Comment:
do we really need this member-variable?

Reply:
I guess not, it was already in the Powell class and I kept it for consistency.

Reply:
Ah, ok. If not it exists in both classes PSO and Powell, maybe we could move to the parent class and then decide if this variable really needed.